lib/import/mobility.js

```diff
@@ -0,0 +1,209 @@
+      if (check_row_is_valid(data)) {
+        var o = {};
+        o.date = yyyymmdd_to_iso(data.DATE);
+        o.origin_country_code = country_code;
+        o.destination_country_code = country_code;
+        o.origin_admin_code = country_code + '-' + data.ORIGIN;
+        o.destination_admin_code = country_code + '-' + data.DESTINATION;
+        o.count = data.COUNT;
+        mobility_objects.push(new Mobility(o));
+      }
+      if (verbose && count % 1000 === 0) {console.log(count);}
+      ++count;
+    })
+    .on('end', function() {
+      bulk_save(mobility_objects, verbose)
+      .then(function(err) {
+        if (err) {return reject(err);}
+        resolve();
+      });
+    });
+  });
+}
+
+/**
+ * Import mobility collection into mongo.
+ *
+ * @param{array} ary - array of mobility objects
+ * @param{bool} verbose - whether to log messages while importing.
+ * @return{promise} Fulfilled once all records are saved.
+ */
+function bulk_save(ary, verbose) {
+  var debug_msg = 'Have left to save';
+  return new Promise(function(resolve, reject) {
+  /**
+   * Save a splice of array of mobility records.
+   * @param{array} ary - array of mobility objects
+   */
+    function save(ary) {
+      if (ary.length > 0) {
+        if (verbose) { console.log(debug_msg, ary.length);}
+        Mobility.insertMany(ary.splice(0, 1000))
+        .then(function() {
+          save(ary);
+        })
+        .catch(function(err) {
+          return reject(err);
+        });
+      } else {
+        resolve();
+      }
+    }
+    save(ary);
+  });
+}
+
+/**
+ * Return list of mobility files in directory.
+ *
+ * @param{string} dir - directory where mobility files are stored
+ * @return{promise} Fulfilled once all files are in list
+ */
+function get_mobility_files(dir) {
+  return new Promise(function(resolve, reject) {
+    fs.readdir(dir, function(err, list) {
+      if (err) { console.log(err); return reject(err);}
+      list = list.filter(function(e) {
+        return e.match(/.csv/);
+      });
+      resolve(list);
+    });
+  });
+}
+
+/**
+ * Import mobility from CSV to Mongo.
+ *
+ * @param{string} dir - directory where mobility csv files exist.
+ * @param{string} country_code - lowercase ISO 3166-1 alpha-2 country code
+ *   https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2
+ * @param{bool} verbose - whether to log messages while importing.
+ * @return{promise} Fulfilled once all records are saved.
+ *
+ * Needs to be connected to mongo before calling.
+ */
+function import_mobility(dir, country_code, verbose) {
+  var all_done = [];
+  return get_mobility_files(dir)
+  .then(function(list) {
+    list.forEach(function(csv_file) {
+      var promise = new Promise(function(resolve) {
+        import_file(dir + '/' + csv_file, country_code, verbose)
+        .then(function() {
+          resolve();
+        });
+      });
+      all_done.push(promise);
+    });
+    return Promise.all(all_done);
+  });
+}
+
```

Comment:
Yep, I ran into issues with the fake weather generator because this `Promise.all` pattern isn't the best for this. Check out these docs (they're from the `Q` docs, but they're not specific to Q at all):
https://github.com/kriskowal/q#sequences

The `reduce` is kind of nice, though I just do a `forEach` in the fake weather code:
https://github.com/unicef/majicbox/blob/master/lib/import/fake_weather.js#L47

Basically, get rid of the `all_done` array and the `Promise.all`. Instead, you set up a single Promise object, then iterate over the CSV files, then use `then` on that Promise object, resetting the value each time, which causes the loop to happen in sequence:
```
var result = Promise.resolve();
get_mobility_files(dir).then(function(files) {
  files.forEach(function(file) {
    result = result.then(function() {
      return import_file(dir + '/' + csv_file, country_code, verbose);
    });
  });
});
return result;
```